File: javascript/step04/test02.js
/* 주제: 함수 - 정의와 사용 II
*/
// 1) 파라미터(X), 리턴 값(X)

function f1() {
  console.log('f1() 호출됨');
}

f1(); // 함수를 호출할 때는 값을 넘기지 않더라도 괄호를 반드시 써야 한다.

// 2) 파라미터(X), 리턴 값(O)
function f2() {
  return 'hello'; // 값을 리턴하려면 return 명령을 사용해야 한다.
}

var value = f2();
console.log('f2()의 리턴 값 = ', value);
// console.log(f2());

// 3) 파라미터(O), 리턴 값(X)
function f3(a) {
  console.log("===>", a);
}

f3(3); // 함수를 호출할 때 값을 전달하고 싶다면 괄호 안에 순서대로 넣어라.
f3(300);
f3("실컷 호출");

function f4(a) {
  return 'a = ' + a + "입니다.";
}

var value = f4(100);
console.log(value);

// 굳이 값을 별도의 변수에 저장한 후 출력할 필요는 없다.
// 함수의 리턴 값을 즉시 사용할 수 있다.
// 참고! 실행은 항상 괄호() 안쪽부터 실행한다.
console.log(f4(500)); // 항상 괄호안부터 실행된다. 즉, f4(500)가 실행되고 console.log()가 실행된다.

File: java01/src/step14/Test02.java
/* 제네릭(Generic) 문법 사용 후 
 * => 값을 꺼낼 때마다 본래 타입으로 형 변환할 필요없다. 
 * 특정 타입의 값만 다루도록 제한할 수 있다. */
package step14;

import java.util.Date;

public class Test02 {

  public static void main(String[] args) {
    // 1) 제네릭 클래스 사용시 타입지정하지 않으면 기본이 Object
    Bucket2 b = new Bucket2();
//    Bucket2<Object> b = new Bucket2<Object>(); // 상동
    
    b.setValue(new String("Hello"));
    String str = (String)b.getValue();
    
    b.setValue(new Date());
    Date date = (Date)b.getValue();
    
    b.setValue(new Integer(10));
    Integer i = (Integer)b.getValue();
  }

}
